nectar/downloaders/threaded.py

```diff
@@ -200,9 +208,19 @@ def _fetch(self, request, session):
             if netloc in self.failed_netlocs:
                 raise SkipLocation()
 
-            response = session.get(
-                request.url, headers=headers, timeout=(self.config.connect_timeout, self.config.read_timeout)
-            )
+            for attempt in xrange(self.tries):
+                try:
+                    response = session.get(
+                        request.url, headers=headers, timeout=(self.config.connect_timeout,
+                                                               self.config.read_timeout)
+                    )
+                    break
+                except requests.ConnectionError as e:
+                    if isinstance(e.strerror, httplib.BadStatusLine):
```

Comment:
I would log here that the failure happened. If a retry succeeds, we still want the user to know that there might be a problem with the remote endpoint.